app/scripts/lib/nonce-tracker.js

```diff
@@ -4,11 +4,17 @@ const Mutex = require('await-semaphore').Mutex
 
 class NonceTracker {
 
-  constructor ({ provider, getPendingTransactions, getConfirmedTransactions }) {
+  constructor ({
+    provider,
+    getPendingTransactions,
+    getConfirmedTransactions,
+    getNetwork,
```

Comment:
instead of passing getNetwork add it as a filter param here: https://github.com/MetaMask/metamask-extension/blob/master/app/scripts/controllers/transactions.js#L30L35

Reply:
however i could be convinced otherwise